libminifi/src/core/yaml/CheckRequiredField.cpp

```diff
@@ -27,28 +28,48 @@ namespace minifi {
-      errMsg =
-          name_node ?
-              "Unable to parse configuration file for component named '" + name_node.as<std::string>() + "' as required field '" + fieldName + "' is missing" :
-              "Unable to parse configuration file as required field '" + fieldName + "' is missing";
-      if (!yamlSection.empty()) {
-        errMsg += " [in '" + yamlSection + "' section of configuration file]";
-      }
-      const YAML::Mark mark = yamlNode->Mark();
-      if (!mark.is_null()) {
-        errMsg += " [line:column, pos at " + std::to_string(mark.line) + ":" + std::to_string(mark.column) + ", " + std::to_string(mark.pos) + "]";
-      }
+bool isFieldPresent(const YAML::Node &yaml_node, std::string_view field_name) {
+  return bool{yaml_node.as<YAML::Node>()[field_name.data()]};
+}
+
+std::string buildErrorMessage(const YAML::Node &yaml_node, const std::vector<std::string> &alternate_field_names, std::string_view yaml_section) {
+  const YAML::Node name_node = yaml_node.as<YAML::Node>()["name"];
+  // Build a helpful error message for the user so they can fix the
+  // invalid YAML config file, using the component name if present
+  auto field_list_string = utils::StringUtils::join(", ", alternate_field_names);
+  std::string err_msg =
+      name_node ?
+          "Unable to parse configuration file for component named '" + name_node.as<std::string>() + "' as none of the possible required fields [" + field_list_string + "] is available" :
+          "Unable to parse configuration file as none of the possible required fields [" + field_list_string + "] is available";
+  if (!yaml_section.empty()) {
+    err_msg += " [in '" + std::string(yaml_section) + "' section of configuration file]";
+  }
+  const YAML::Mark mark = yaml_node.Mark();
+  if (!mark.is_null()) {
+    err_msg += " [line:column, pos at " + std::to_string(mark.line) + ":" + std::to_string(mark.column) + ", " + std::to_string(mark.pos) + "]";
+  }
+  return err_msg;
+}
+
+void checkRequiredField(const YAML::Node &yaml_node, std::string_view field_name, std::string_view yaml_section, std::string_view error_message) {
+  if (!isFieldPresent(yaml_node, field_name)) {
+    if (error_message.empty()) {
+      throw std::invalid_argument(buildErrorMessage(yaml_node, std::vector<std::string>{std::string(field_name)}, yaml_section));
     }
-    logger->log_error(errMsg.c_str());
-    throw std::invalid_argument(errMsg);
+    throw std::invalid_argument(std::string(error_message));
+  }
+}
+
+std::string getRequiredField(const YAML::Node &yaml_node, const std::vector<std::string> &alternate_names, std::string_view yaml_section, std::string_view error_message) {
+  for (const auto& name : alternate_names) {
+    if (yaml::isFieldPresent(yaml_node, name)) {
+      return yaml_node[name].as<std::string>();
+    }
+  }
+  if (error_message.empty()) {
+    std::invalid_argument(buildErrorMessage(yaml_node, alternate_names, yaml_section));
```

Comment:
I think you meant to `throw` this.

Reply:
Fixed in e6d61d1d6b803061bc6dd60ec9800109e789efda